src/search/landmarks/util.cc

```diff
@@ -79,67 +79,79 @@ int get_operator_or_axiom_id(const OperatorProxy &op) {
     }
 }
 
-static void dump_node(const TaskProxy &task_proxy, const LandmarkNode &node) {
-    cout << "  lm" << node.get_id() << " [label=\"";
-    bool first = true;
-    const Landmark &landmark = node.get_landmark();
-    for (FactPair fact : landmark.facts) {
-        if (!first) {
-            if (landmark.disjunctive) {
-                cout << " | ";
-            } else if (landmark.conjunctive) {
-                cout << " & ";
+static void dump_node(
+    const TaskProxy &task_proxy,
+    const LandmarkNode &node,
+    utils::LogProxy &log) {
+    if (log.is_at_least_debug()) {
```

Comment:
This is what we decided in the meeting two days ago: every logging must be guarded. Someone could just call dump_node from outside dump_landmark_graph.

Reply:
Exactly, it will go away in the redesign. Having the `at_least_debug` right here will help us in that revision to figure out which method we need to call (such as `log.debug`) without having to think. In the redesign, there won't be a way to log without specifying the level of the log message, so we will need to know then what the log level of the message is.

Reply:
Will there be an option to still exclude the `dump` functions from outside? Because otherwise this would still loop over the entire landmark graph just to find that it should only log this in debug mode.

Reply:
Log level tests will still be possible everywhere. I'm not sure what you mean with "this"; this function in paticular doesn't loop over landmark nodes, and if the log level isn't at least debug, it will exit immediately. But if there is a caller that triggers a lot of logging, of course it can include its own guards.

For what it's worth, a name like `dump_...` or `print_...` doesn't work that well with the new model because I think there would be an expectation that it prints unconditionally. Probably a name like `log_...` would be better.

Reply:
Perhaps it's worth adding a section on logging to ForDevelopers, either as part of the coding conventions or on a subpage to combine these things.